src/rouge/worker/worker.py

```
@@ -157,26 +247,70 @@ def execute_workflow(self, issue_id: int, description: str) -> bool:
-                update_issue_status(issue_id, "pending", self.logger)
-                return False
+                update_patch_status(patch_id, "failed", self.logger)
+                transition_to_patch_pending(issue_id)
+                return patch_wf_id, False
+
+        except (ValueError, subprocess.TimeoutExpired, Exception) as e:
+            # If we have a patch_id, mark the patch as failed
+            if patch_id is not None:
+                self.logger.error(
+                    f"Patch workflow failed for issue {issue_id}, patch {patch_id}: {e}"
+                )
```

Ruff (0.14.11)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)